packages/drivers/odsp-driver/src/contractsPublic.ts

```diff
@@ -5,6 +5,40 @@
 
 import { IOdspUrlParts } from "@fluidframework/odsp-driver-definitions";
 
+/**
+ * Socket storage discovery api response
+ */
+export interface ISocketStorageDiscovery {
+	// The id of the web socket
+	id: string;
+
+	// SPO gives us runtimeTenantId, we remap it to tenantId
+	// See getSocketStorageDiscovery
+	runtimeTenantId?: string;
+	tenantId: string;
+
+	snapshotStorageUrl: string;
+	deltaStorageUrl: string;
+
+	/**
+	 * PUSH URL
+	 */
+	deltaStreamSocketUrl: string;
+
+	/**
+	 * The access token for PushChannel. Optionally returned, depending on implementation.
+	 * OneDrive for Consumer implementation returns it and OneDrive for Business implementation
+	 * does not return it and instead expects token to be returned via `getWebsocketToken` callback
+	 * passed as a parameter to `OdspDocumentService.create()` factory.
+	 */
+	socketToken?: string;
```

Comment:
Given that we touch this area, could you please follow up with Marcelo if this thing should be deleted (or maybe vice versa - become non-optional)? I do not remember the history well, but I think it was temp thing, maybe it's time to clean it (if we can)

Reply:
I will follow up on this separately. Seems like when getWebsocketToken callback is not supplied by the app, then we look for the socket token from join session. I am not sure whether they always supply this or not.